auth/service.go

```diff
@@ -560,6 +560,24 @@ func (svc service) CreateDomain(ctx context.Context, token string, d Domain) (do
 	return dom, nil
 }
 
+func (svc service) DeleteDomain(ctx context.Context, token string, d Domain) error {
+	key, err := svc.Identify(ctx, token)
+	if err != nil {
+		return errors.Wrap(svcerr.ErrAuthentication, err)
+	}
+	if err := svc.Authorize(ctx, PolicyReq{
+		Subject:     key.Subject,
+		SubjectType: UserType,
+		SubjectKind: UsersKind,
+		ObjectType:  DomainType,
+		Permission:  AdminPermission,
+		Object:      d.ID,
+	}); err != nil {
+		return errors.Wrap(svcerr.ErrAuthorization, err)
+	}
+	return svc.domains.Delete(ctx, d.ID)
```

Comment:
```go
if err := svc.domains.Delete; err != nil {
    return errors.Wrap(svcerr.ErrFailedDelete, err)
}
```

We should not return the repo error directly since it may leak some db info which is not desirable.